UITests/FindInPageTests.swift

```diff
@@ -0,0 +1,107 @@
+        clickFindNext()
+        try! tester().tryFindingViewWithAccessibilityLabel("2/4")
+        clickFindNext()
+        try! tester().tryFindingViewWithAccessibilityLabel("3/4")
+        clickFindNext()
+        try! tester().tryFindingViewWithAccessibilityLabel("4/4")
+        clickFindNext()
+        try! tester().tryFindingViewWithAccessibilityLabel("1/4")
+        clickFindPrevious()
+        try! tester().tryFindingViewWithAccessibilityLabel("4/4")
+        clickFindPrevious()
+        try! tester().tryFindingViewWithAccessibilityLabel("3/4")
+        clickFindPrevious()
+        try! tester().tryFindingViewWithAccessibilityLabel("2/4")
+        clickFindPrevious()
+        try! tester().tryFindingViewWithAccessibilityLabel("1/4")
+
+        // Test a query with no matches.
+        let findTextField = tester().waitForViewWithAccessibilityValue("nullam") as! UITextField
+        findTextField.becomeFirstResponder()
+        tester().enterTextIntoCurrentFirstResponder("z")
+        let resultsView = tester().waitForViewWithAccessibilityLabel("0/0")
+        XCTAssertFalse(resultsView.hidden)
+        tester().clearTextFromFirstResponder()
+        XCTAssertTrue(resultsView.hidden)
+
+        // Make sure the selection menu still works with the bar already visible.
+        openFindInPageBar(webView)
+        try! tester().tryFindingViewWithAccessibilityLabel("1/4")
+
+        // Make sure the bar disappears when reloading.
+        tester().tapViewWithAccessibilityLabel("Reload")
+        tester().waitForAbsenceOfViewWithAccessibilityLabel("nullam")
+
+        // Make sure the bar disappears when opening the tabs tray.
+        openFindInPageBar(webView)
+        tester().tapViewWithAccessibilityLabel("Show Tabs")
+        tester().tapViewWithAccessibilityLabel("Find Page")
+        tester().waitForAbsenceOfViewWithAccessibilityLabel("nullam")
+
+        // Test that Done dismisses the toolbar.
+        openFindInPageBar(webView)
+        tester().tapViewWithAccessibilityLabel("Done")
+        XCTAssertFalse(viewExists("Done"))
+        XCTAssertFalse(viewExists("Next in-page result"))
+        XCTAssertFalse(viewExists("Previous in-page result"))
+    }
+
+    private func openFindInPageBar(webView: WKWebView) {
+        // Make the selection menu appear. To keep things simple, the page has absolutely
+        // positioned text at the top-left corner.
+        webView.longPressAtPoint(CGPointZero, duration: FindInPageTests.LongPressDuration)
+        webView.tapAtPoint(CGPointZero)
+        tester().tapViewWithAccessibilityLabel("Find in Page")
+        tester().waitForViewWithAccessibilityValue("nullam")
+    }
+
+    private func clickFindNext() {
+        tester().tapViewWithAccessibilityLabel("Next in-page result")
+    }
+
+    private func clickFindPrevious() {
+        tester().tapViewWithAccessibilityLabel("Previous in-page result")
+    }
+
+    private func viewExists(label: String) -> Bool {
```

Comment:
I like this method :+1: I keep finding myself using something like this in other UITests. Could we move it into a Utils class or tester extension?